src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,29 +1,130 @@
+      marginRight: 'auto',
+      width: '310.71px',
+    },
+  });
+
+interface UserProps {
+  classes: { title: string; headerWrapper: string; selectionHeader: string; scrollDiv: string };
+}
+
+function CustomerMain(props: RouteComponentProps & UserProps) {
   useEffect(() => {
     getCustomers();
   }, []);
 
+  const { classes } = props;
   const [customers, setCustomers] = useState<CustomerRecord[]>([]);
+  const [allCustomersTrie, setAllCustomersTrie] = useState<TrieTree<CustomerRecord>>(new TrieTree('name'));
+  const [fullCustomers, setFullCustomers] = useState<CustomerRecord[]>([]);
 
-  const getCustomers = async () => {
-    const customers: CustomerRecord[] = await getAllCustomers();
-    setCustomers(customers);
-  };
+  const getCustomers = () => {
+    const siteData = store.getState().siteData;
+    const sites: SiteRecord[] = siteData.sites;
+    let allCustomers: CustomerRecord[] = [];
+    for (let i = 0; i < sites.length; i++) {
+      let currCustomers: CustomerRecord[] = sites[i].customers;
+      allCustomers = allCustomers.concat(currCustomers)
+    }
+    setCustomers(allCustomers);
+    setFullCustomers(allCustomers);
+    console.log(allCustomers)
+    const customersTrie: TrieTree<CustomerRecord> = new TrieTree('name');
+    customersTrie.addAll(allCustomers);
+    setAllCustomersTrie(customersTrie);
 
+  }
+
+  const calcLatestReadingDate = (curr: CustomerRecord) => {
+    //depends if the meter readings list is sorted with earliest => latest
+    let latestMeterReading = curr.meterReadings[curr.meterReadings.length - 1]
+    const dateTime = Date.parse(latestMeterReading.date)
+    let readingDate = new Date(dateTime)
+    let month = readingDate.getMonth() + 1;
+    let day = readingDate.getDate();
+    let shortDate = month.toString() + '.' + day.toString();
+    // console.log(readingDate.toLocaleDateString())
+    // console.log(shortDate)
+    return shortDate;
+  }
+
+  const handleSearchChange = (e: any) => {
+    const searchVal = e.target.value.trim();
+    if (searchVal === '') {
+      setCustomers(fullCustomers);
+      return;
+    }
+    const customers = allCustomersTrie.get(searchVal);
+    setCustomers(customers);
+  }
   return (
-    <>
-      {customers.map((customer, index) => (
-        <Link key={index} to={{ pathname: `${props.match.url}/${customer.name}`, state: { customer: customer } }}>
-          <h3>{customer.name}</h3>
-        </Link>
-      ))}
-    </>
+    <div>
+      <div className={classes.headerWrapper}>
+        <h1 className={classes.title}>Customers</h1>
+        <div className={classes.selectionHeader}>
+          <UserSearchBar searchFun={handleSearchChange} />
+          <FormControl>
+            <Select inputProps={{ 'aria-label': 'Without label' }}>
+              {/* placeholder sorting for now */}
+              <MenuItem value={10}>Ten</MenuItem>
+              <MenuItem value={20}>Twenty</MenuItem>
+              <MenuItem value={30}>Thirty</MenuItem>
```

Comment:
Can we get this `Sort By` working in this PR? Is there a reason it's placeholdered / on backburner?

Reply:
I think Jen originally gave the task to Emma? So I didn't implement it. I can do it if needed.